MailerQ/ApiClient/ApiClient.cs

```diff
@@ -15,186 +17,200 @@ public class ApiClient : IApiClient
                 return result == HttpStatusCode.OK;
             }
             catch (Exception exception)
             {
                 _logger.LogError(exception,
                     "Endpoint: {Endpoint}. Method: {Method}.",
-                    EndpointPause,
-                    MethodPost);
+                    pause.Endpoint,
+                    nameof(Post));
                 throw exception;
             }
         }
 
-        public bool Post(Inject inject)
+        public async Task<bool> Post(Inject inject)
         {
             try
             {
                 var jsonContent = JsonConvert.SerializeObject(inject);
-                var result = MailerQApiClient(EndpointInject, MethodPost, jsonContent);
+                var result = await MailerQApiClientPostAsync(inject.Endpoint, jsonContent);
                 return result == HttpStatusCode.OK;
             }
             catch (Exception exception)
             {
                 _logger.LogError(exception,
                     "Endpoint: {Endpoint}. Method: {Method}.",
-                    EndpointInject,
-                    MethodPost);
+                    inject.Endpoint,
+                    nameof(Post));
                 throw exception;
             }
         }
-        public bool Delete(Error error)
+
+        private async Task<HttpStatusCode> MailerQApiClientPostAsync(string endpoint, string content)
+        {
+            _apiUri.Path = Path.Combine(_apiUri.Path, endpoint);
+
+            var client = _httpFactory.CreateClient();
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", _apiClientSettings.Token);
+
+            HttpContent httpContent = new StringContent(content);
+            httpContent.Headers.ContentType.MediaType = "application/json";
+
+            var httpResponse = await client.PostAsync(_apiUri.Uri, httpContent);
+
+            return httpResponse.StatusCode;
+        }
+
+        public async Task<bool> Delete(Error error)
         {
             try
             {
                 var jsonContent = JsonConvert.SerializeObject(error);
-                var result = MailerQApiClient(EndpointError, MethodDelete, jsonContent);
+                var result = await MailerQApiClientDeleteAsync(error.Endpoint, jsonContent);
                 return result == HttpStatusCode.OK;
             }
             catch (Exception exception)
             {
                 _logger.LogError(exception,
                     "Endpoint: {Endpoint}. Method: {Method}.",
-                    EndpointError,
-                    MethodDelete);
+                    error.Endpoint,
+                    nameof(Delete));
                 throw exception;
             }
         }
-        public bool Delete(Pause pause)
+        public async Task<bool> Delete(Pause pause)
         {
             try
             {
                 var jsonContent = JsonConvert.SerializeObject(pause);
```

Comment:
What do you think about moving this serialization to the same place where you set the content type?